{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "dd51bd15-2c47-43b0-8238-a0e78618f012",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import os\n",
    "from Bio import SeqIO, AlignIO\n",
    "import pysam\n",
    "# from Bio.Align.Applications import PrankCommandline\n",
    "# import subprocess\n",
    "from collections import Counter\n",
    "import statistics"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "b0bf7e3f-d8fc-4e16-b748-e69b789b2c30",
   "metadata": {},
   "outputs": [],
   "source": [
    "def groupUMI(ucfile):\n",
    "    uc = pd.read_csv(ucfile,delimiter=\"\\t\",header=None,usecols=[0,1,8])\n",
    "    uc.columns=[\"a\",\"cluster\",\"name\"]\n",
    "    uc = uc.loc[uc[\"a\"].isin([\"C\",\"H\"])]\n",
    "    # uc.index=uc[\"name\"]\n",
    "    uc = uc.loc[uc.duplicated(subset='cluster', keep=False)]\n",
    "    return(uc)\n",
    "\n",
    "\n",
    "def dictBam(bamfile, outDict):\n",
    "    \"\"\"\n",
    "    i\n",
    "    \"\"\"\n",
    "    with pysam.AlignmentFile(bamfile, \"rb\", check_sq=False) as infile:\n",
    "        with open(outDict, \"w\") as outfile:\n",
    "#         dic ={}\n",
    "            for read in infile.fetch(until_eof=True):\n",
    "                dic = {}\n",
    "                if not read.has_tag(\"MD\") or (query_seq := read.query_sequence) is None or read.is_secondary:\n",
    "                    return None\n",
    "\n",
    "                readDict = callMutation(read)\n",
    "                dic[read.qname] = readDict\n",
    "                outfile.write(f\"{read.qname}\\t{readDict[0]}\\t{readDict[1]}\\t{readDict[2]}\\t{readDict[3]}\\n\")\n",
    "\n",
    "def callMutation(read):\n",
    "    \"\"\"\n",
    "    get mutations and indels from a read which has MD tag in sam/bam\n",
    "    \"\"\"\n",
    "\n",
    "    mut_type = []\n",
    "    mut_qual= []\n",
    "    indel_type = []\n",
    "    indel_qual =[]\n",
    "\n",
    "    ref_index = read.reference_start-1\n",
    "    query_index = -1\n",
    "    I_index = [0]\n",
    "    D_index = [0]\n",
    "\n",
    "    for query_pos, ref_pos, ref_base in read.get_aligned_pairs(with_seq=True): #matches_only=True, # if True, no None on either side\n",
    "\n",
    "        if query_pos is None: # Deletion\n",
    "            ref_index += 1\n",
    "            D_index.append(ref_index)\n",
    "            if D_index[-1]-D_index[-2] >1:\n",
    "                indel_type.append(str(ref_index)+\"D\")\n",
    "                indel_qual.append(read.query_qualities[query_index])\n",
    "\n",
    "        elif ref_pos is None: # Insertion\n",
    "            query_index +=1\n",
    "            I_index.append(ref_index)\n",
    "            if I_index[-1]-I_index[-2] >1:\n",
    "                indel_type.append(str(ref_index)+\"I\")\n",
    "                indel_qual.append(read.query_qualities[query_pos])\n",
    "\n",
    "        elif ref_base.islower():\n",
    "            ref_index += 1\n",
    "            query_index +=1\n",
    "            mut_type.append(str(ref_pos)+ref_base.upper()+read.query_sequence[query_pos])\n",
    "            mut_qual.append(read.query_qualities[query_pos])\n",
    "\n",
    "        else:\n",
    "            ref_index += 1\n",
    "            query_index +=1\n",
    "\n",
    "    return(mut_type, mut_qual, indel_type, indel_qual)\n",
    "\n",
    "\n",
    "# def generateConsensus(dictbam, uc, my_cluster, frequency_threshold = 0.5, quality_threshold = 4):\n",
    "#     uc = groupUMI(ucfile)\n",
    "#     my_cluster = 2\n",
    "#     clusters = ucsam.cluster.unique()\n",
    "#     for my_cluster in clusters:\n",
    "#         sub=uc.loc[uc[\"cluster\"]==my_cluster]\n",
    "#         if len(sub)>=2:\n",
    "#             ssMuts = ccSeq(dictbam, sub, frequency_threshold, quality_threshold)\n",
    "            \n",
    "            \n",
    "\n",
    "def ccSeq(sub, frequency_threshold, quality_threshold):\n",
    "    \"\"\"\n",
    "    find cluster consensus mutations in each uc group\n",
    "    \"\"\"\n",
    "    num = len(sub)\n",
    "    ccMuts = []\n",
    "    \n",
    "    muts =sub[\"mut_type\"].to_numpy()\n",
    "    muts = removeBracket(muts, qual = False)\n",
    "    mut_quals =sub[\"mut_qual\"].to_numpy()\n",
    "    mut_quals = removeBracket(mut_quals, qual = True)\n",
    "    indels =sub[\"indel_type\"].to_numpy()\n",
    "    indels = removeBracket(indels, qual = False)\n",
    "    indel_quals =sub[\"indel_qual\"].to_numpy()\n",
    "    indel_quals = removeBracket(indel_quals, qual = True)\n",
    "      \n",
    "    fM = filterMuts(muts, mut_quals, num, frequency_threshold, quality_threshold)\n",
    "    fID = filterMuts(indels, indel_quals, num, frequency_threshold, quality_threshold)\n",
    "    ccMuts.append(fM)\n",
    "    ccMuts.append(fID)\n",
    "    ccMuts = flatList(ccMuts)\n",
    "    return(ccMuts)\n",
    "\n",
    "def removeBracket(s,qual=True):\n",
    "    \"\"\"\n",
    "    ori: [\"['2048TC']\" '[]' \"['1283CT', '1529GC']\"]\n",
    "    transformed: ['2048TC','1283CT','1529GC']\n",
    "    \"\"\"\n",
    "    su = []\n",
    "    if s is not None:\n",
    "        for s_sub in s:\n",
    "            s_sub = s_sub.replace(\"[\",\"\")\n",
    "            s_sub = s_sub.replace(\"]\",\"\")\n",
    "            s_sub = s_sub.replace(\"'\",\"\")\n",
    "            s_sub = s_sub.split(\",\")\n",
    "            for s_sub_sub in s_sub:\n",
    "                if s_sub_sub != '':\n",
    "                    s_sub_sub = s_sub_sub.replace(\" \",\"\")\n",
    "                    if qual:\n",
    "                        su.append(int(s_sub_sub))\n",
    "                    else:\n",
    "                        su.append(str(s_sub_sub))\n",
    "    return(su)\n",
    "    \n",
    "def filterMuts(muts, mut_quals, num, frequency_threshold = 0.5, quality_threshold = 4):\n",
    "    \"\"\"\n",
    "    filter mutations/indels with frequency >= frequency_threshold\n",
    "    \"\"\"\n",
    "    cs = [ele for ele, cnt in Counter(muts).items() if cnt >= num*frequency_threshold]\n",
    "    csMuts = []\n",
    "    if len(cs) >0:\n",
    "        for mutType in cs:\n",
    "            quals = []\n",
    "            for b,q in zip(muts, mut_quals):\n",
    "                if b == mutType:\n",
    "                    quals.append(q)\n",
    "            if statistics.mean(quals) >= quality_threshold:\n",
    "                csMuts.append(mutType)\n",
    "    return(csMuts)\n",
    "\n",
    "\n",
    "def flatList(Alist):\n",
    "    flat_list = []\n",
    "    for sublist in Alist:\n",
    "        for item in sublist:\n",
    "            flat_list.append(item)        \n",
    "    return(flat_list)\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "cc9d7065-cdd8-4933-9f60-3fafaa72da24",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4T\n",
      "5T\n",
      "16T\n",
      "47_1T\n",
      "47_4T\n",
      "47_5T\n",
      "47_6T\n",
      "47_8T\n",
      "50T\n",
      "66_1T\n"
     ]
    }
   ],
   "source": [
    "if __name__ == \"__main__\":\n",
    "    \"\"\"\n",
    "    generate consensus\n",
    "    \"\"\"\n",
    "        \n",
    "    SAMPLE_NAME = \"DSST\"\n",
    "    SAMPLE_BARCODES = \"/data/zhaolian/LineageTracing/DSS/PacBio/sampleBarcodes_\"+SAMPLE_NAME+\".txt\"\n",
    "    BAMDIR=\"/data/zhaolian/LineageTracing/DSS/PacBio/5.bam_split/\"\n",
    "    UCDIR=\"/data/zhaolian/LineageTracing/DSS/PacBio/4.umi_clustering/\"\n",
    "    OUTDIR=\"/data/zhaolian/LineageTracing/DSS/PacBio/6.umiConsensus/\"\n",
    "    numUC=3 ## number of ccs reads in one consensus cluster\n",
    "    mutFreq=0.8 ## export the mutation if mutation frequency >= mutFreq \n",
    "\n",
    "    if not os.path.exists(OUTDIR):\n",
    "        os.mkdir(OUTDIR)\n",
    "\n",
    "    with open(SAMPLE_BARCODES,\"r\") as f_barcode:\n",
    "        for line in f_barcode.readlines():\n",
    "            sample = line.replace(\"\\n\", \"\").split(\"\\t\")[0]\n",
    "            print(sample)\n",
    "\n",
    "            bamfile = BAMDIR+sample+\".bam\"\n",
    "            outDict = OUTDIR+sample+\"_mutation_per_read\"+str(numUC)+\"_\"+str(mutFreq)+\".txt\" ## output file 1\n",
    "            dictBam(bamfile, outDict)\n",
    "\n",
    "            ucfile = UCDIR+sample+\"_umi_UsearchClusters.uc\"\n",
    "            outfile = OUTDIR+sample+\"_umiConsensus\"+str(numUC)+\"_\"+str(mutFreq)+\".tsv\" ## output file 2\n",
    "\n",
    "            uc = groupUMI(ucfile)\n",
    "            colnames =[\"name\",\"mut_type\", \"mut_qual\", \"indel_type\", \"indel_qual\"]\n",
    "            dictbam = pd.read_csv(outDict,delimiter=\"\\t\",header=None,names=colnames)\n",
    "            ucbam=pd.merge(uc,dictbam, on=\"name\",how=\"inner\")\n",
    "\n",
    "\n",
    "            clusters = uc.cluster.unique()\n",
    "            with open(outfile,\"w\") as f:\n",
    "                for(my_cluster) in clusters:\n",
    "                    sub=ucbam.loc[ucbam[\"cluster\"]==my_cluster]\n",
    "                    if len(sub) >= 3:\n",
    "                        ssMuts = ccSeq(sub, frequency_threshold = 0.8, quality_threshold = 4)\n",
    "                        if len(ssMuts) >0:\n",
    "                            s=len(ssMuts)\n",
    "    #                         print(ssMuts)\n",
    "                            f.write(f\"{my_cluster}\\t{s}\\t{','.join([i for i in ssMuts])}\\n\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2cb2402e-25cd-4003-911f-a2f1f9d02a7d",
   "metadata": {},
   "outputs": [],
   "source": [
    "# SAMPLE_NAME = \"DSST\"\n",
    "# SAMPLE_BARCODES = \"/data/zhaolian/LineageTracing/DSS/PacBio/sampleBarcodes_\"+SAMPLE_NAME+\".txt\"\n",
    "BAMDIR=\"/data/zhaolian/LineageTracing/DSS/PacBio/5.bam_split/\"\n",
    "UCDIR=\"/data/zhaolian/LineageTracing/DSS/PacBio/4.umi_clustering/\"\n",
    "OUTDIR=\"/data/zhaolian/LineageTracing/DSS/PacBio/6.umiConsensus/\"\n",
    "numUC=3 ## number of ccs reads in one consensus cluster\n",
    "mutFreq=0.8 ## export the mutation if mutation frequency >= mutFreq \n",
    "\n",
    "print(sample)\n",
    "\n",
    "bamfile = BAMDIR+sample+\".bam\"\n",
    "outDict = OUTDIR+sample+\"_mutation_per_read\"+str(numUC)+\"_\"+str(mutFreq)+\".txt\" ## output file 1\n",
    "dictBam(bamfile, outDict)\n",
    "\n",
    "ucfile = UCDIR+sample+\"_umi_UsearchClusters.uc\"\n",
    "outfile = OUTDIR+sample+\"_umiConsensus\"+str(numUC)+\"_\"+str(mutFreq)+\".tsv\" ## output file 2\n",
    "\n",
    "uc = groupUMI(ucfile)\n",
    "colnames =[\"name\",\"mut_type\", \"mut_qual\", \"indel_type\", \"indel_qual\"]\n",
    "dictbam = pd.read_csv(outDict,delimiter=\"\\t\",header=None,names=colnames)\n",
    "ucbam=pd.merge(uc,dictbam, on=\"name\",how=\"inner\")\n",
    "\n",
    "\n",
    "clusters = uc.cluster.unique()\n",
    "with open(outfile,\"w\") as f:\n",
    "    for(my_cluster) in clusters:\n",
    "        sub=ucbam.loc[ucbam[\"cluster\"]==my_cluster]\n",
    "        if len(sub) >= 3:\n",
    "            ssMuts = ccSeq(sub, frequency_threshold = 0.8, quality_threshold = 4)\n",
    "            if len(ssMuts) >0:\n",
    "                s=len(ssMuts)\n",
    "#                         print(ssMuts)\n",
    "                f.write(f\"{my_cluster}\\t{s}\\t{','.join([i for i in ssMuts])}\\n\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3ec921d8-91f8-4985-9a32-b8ec6505936d",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a23506b7-5ecc-456a-80b5-6b3a9997862f",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "229ee5f0-d6d2-46c5-b319-98b42feed4fb",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2156470c-931a-4a8e-8e1b-e7ff637fe46d",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "20b84986-d6e9-4124-9eef-ffdc52dd59a7",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "42e76650-84d6-4b7d-90e5-966119a0541b",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e40df34a-f97e-456f-a94d-e4ed04f2c4f9",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1c3241f8-e775-4e98-9178-7d8718a99581",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "51be3e05-7e9c-4205-802e-e4bb28fa118b",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "36956e25-bcce-4691-bbd6-adb181c100bf",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6439ad4c-94ca-4e59-ba54-80c23eb12bc8",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "24d8e33e-cd71-4968-bec9-13b99ae7ae17",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "980971ee-7816-4e75-96cc-f50408e3bbfd",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "58d19f80-2de6-4c8e-af91-7ad7687f04db",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
